src/modules/name-manager/PermanentNameModule.js

```diff
@@ -0,0 +1,371 @@
+import OldEnsAbi from './ABI/oldEnsAbi.js';
+import OldDeedAbi from './ABI/oldDeedAbi.js';
+import { uploadFileToIpfs, getHashFromFile } from './services';
+import contentHash from 'content-hash';
+import BigNumber from 'bignumber.js';
+import NameManagerInterface from './interface/NameManagerInterface';
+import * as nameHashPckg from 'eth-ens-namehash';
+
+const OLD_ENS_ADDRESS = '0x6090a6e47849629b7245dfa1ca21d94cd15878ef';
+const BURNER_ADDRESS = '0x0000000000000000000000000000000000000000';
+
+export default class PermanentNameModule extends NameManagerInterface {
+  constructor(name, address, network, web3, ens) {
+    super(name, address, network, web3, ens);
+    this.deedValueVal = 0;
+    this.deedOwnerVal = '0x';
+    this.secretPhraseVal = '';
+    this.expirationVal = null;
+    this.expiredVal = false;
+    this.redeemableVal = false;
+    // Contracts
+    this.oldEnsContractVal = null;
+    this.oldDeedContractVal = null;
+    this.dnsRegistrarContractVal = null;
+    this.dnsClaimVal = null;
+    this.dnsStatusVal = '';
+
+    this._initModule();
+  }
+
+  register(duration) {
+    const _self = this;
+    if (this.ownerVal === '0x') {
+      throw new Error('Owner not set! Please initialize module properly!');
+    }
+    return _self
+      ._createCommitment()
+      .then(() => {
+        _self._registerWithDuration(duration);
+      })
+      .then(() => {
+        this._initModule(); // might need something more effecient than this
+      });
```

Comment:
shouldnt this be
```
return _self
      ._createCommitment()
      .then(() => _self._registerWithDuration(duration))
      .then(() => this._initModule()); // might need something more effecient than this
```

Reply:
this way it'll go through the promises and at the end return the promise for this._initModule()